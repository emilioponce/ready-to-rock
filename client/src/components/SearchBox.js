import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { getBandsIdsBySearchTerms } from '../api'

import styles from './SearchBox.module.css'

class SearchBox extends Component {
  constructor(props) {
    super(props)
    this.state = {
      searchTerms: ''
    }
    this.handleClick = this.handleClick.bind(this)
  }

  handleChange = e => {
    this.setState({
      searchTerms: e.target.value
    })
  }

  async handleClick() {
    const { searchTerms } = this.state
    const bandsIds = await getBandsIdsBySearchTerms(searchTerms)
    this.props.onSearchTermChange(bandsIds)
  }

  /* Enables searching by pressing enter key */
  handleKeyPressed = e => {
    var code = e.keyCode || e.which
    if (code === 13) {
      //13 = 'Enter' keycode
      this.handleClick()
    }
  }

  render() {
    const { searchTerms } = this.state
    return (
      <div className={styles.searchBox}>
        <div>
          <input
            type="text"
            className={styles.input}
            placeholder="Enter text .."
            value={searchTerms}
            onChange={this.handleChange}
            onKeyPress={this.handleKeyPressed}
          />
          <button className={styles.button} onClick={this.handleClick}>
            Search
          </button>
        </div>
      </div>
    )
  }
}

SearchBox.propTypes = {
  onSearchTermChange: PropTypes.func.isRequired
}

export default SearchBox
