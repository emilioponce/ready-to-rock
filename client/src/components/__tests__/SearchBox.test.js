import React from 'react'
import ReactDOM from 'react-dom'
import SearchBox from '../SearchBox'
import { shallow } from 'enzyme'

const fn = jest.fn()
const component = <SearchBox onSearchTermChange={fn} />

describe('Component SearchBox', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(component, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('snapshot renders correctly', () => {
    const wrapper = shallow(component)
    expect(wrapper).toMatchSnapshot()
  })

  it('user can introduce text', () => {
    const wrapper = shallow(component)
    const changeEvent = {
      target: { value: 'rock' }
    }
    wrapper.find('input').simulate('change', changeEvent)
    expect(wrapper.find('input').props().value).toEqual('rock')
  })

  it('user introduces text and after a search button click the state changes', () => {
    const wrapper = shallow(component)
    expect(wrapper.state().searchTerms).toBe('')
    const changeEvent = {
      target: { value: 'rock' }
    }
    wrapper.find('input').simulate('change', changeEvent)
    wrapper.find('button').simulate('click')
    expect(wrapper.state().searchTerms).toBe('rock')
  })
})
