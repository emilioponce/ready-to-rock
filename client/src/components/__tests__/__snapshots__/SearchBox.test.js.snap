// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component SearchBox snapshot renders correctly 1`] = `
<div
  className="searchBox"
>
  <div>
    <input
      className="input"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Enter text .."
      type="text"
      value=""
    />
    <button
      className="button"
      onClick={[Function]}
    >
      Search
    </button>
  </div>
</div>
`;
